tasks/gov.js

```diff
@@ -147,47 +100,123 @@ task("gov:setChainlinkPriceFeed")
+        const arrayUniqueByKey = [...new Map(result.map(item =>
+            [item[key], item])).values()];
+
+        // unique addresses regardless of markets 
+        const uniqueAddresses = [...new Set(arrayUniqueByKey.map(item => item.account))];
+
+        // compute health scores
+        let health_scores = {};
+        
+        console.log(`Number of unique addresses to parse in batches: ${uniqueAddresses.length}`);
+        
+        while (uniqueAddresses.length > 0) {
+            console.log("=== Waiting for the next batch to be resolved")
+            const chunkSize = 100;
+            const batch = uniqueAddresses.splice(0, chunkSize);
+            await Promise.all(batch.map(async account => {
+                let status = await ctx.contracts.exec.liquidity(account);
+                let collateralValue = status.collateralValue;
+                let liabilityValue = status.liabilityValue;
+                let healthScore = liabilityValue == 0 ? ethers.constants.MaxUint256 : (collateralValue * et.c1e18) / liabilityValue;
+
+                health_scores[account] = {
+                    health: healthScore / et.c1e18,
+                    collateralValue,
+                    liabilityValue
+                };
+            }));
+            console.log("*** The batch has been resolved")
+        }
+
+        let outputJson = JSON.stringify(health_scores);
+        fs.writeFileSync(`${filename}.json`, outputJson + "\n");
+    });
+
+
+
 task("gov:forkHealthScoreDiff", "Compare the health scores of accounts from a pair of JSON files")
     .addPositionalParam("preGovActionFileName", "file name without .json suffix")
     .addPositionalParam("postGovActionFileName", "file name without .json suffix")
     .addPositionalParam("removeFilesAfterParsing", "true or false")
-    .setAction(async ({preGovActionFileName, postGovActionFileName, removeFilesAfterParsing}) => {
+    .setAction(async ({ preGovActionFileName, postGovActionFileName, removeFilesAfterParsing }) => {
         const fs = require('fs');
-        
+
         const prePath = `${preGovActionFileName}.json`;
         const postPath = `${postGovActionFileName}.json`;
-        
+
         try {
             const pre_gov_scores = require(`../${prePath}`);
             const post_gov_scores = require(`../${postPath}`);
 
             for (let account of Object.keys(pre_gov_scores)) {
+                let collateralValueBefore = ethers.utils.formatEther(pre_gov_scores[account].collateralValue.hex);
+                let liabilityValueBefore = ethers.utils.formatEther(pre_gov_scores[account].liabilityValue.hex);
+                let collateralValueAfter = ethers.utils.formatEther(post_gov_scores[account].collateralValue.hex);
+                let liabilityValueAfter = ethers.utils.formatEther(post_gov_scores[account].liabilityValue.hex);
+                let spyModeURL = `https://app.euler.finance/account/0?spy=${account}`;
+                
+                let result = {
```

Comment:
last thing that would be very convenient here, I think. instead of console logging while processing, push the `result`s into two arrays (one for accounts in violation and one for accounts at risk of liquidation). when all the accounts processed, sort both arrays decreasing by `liabilityValueBefore` and only then display the results one by one (in violation first, then at risk). that way, it'd be easier to assess the impact as the accounts would be sorted from those having biggest liability to the lowest and those in violation would be separated from those at risk.

also, it'd be good to dump the results into a file if requested. would be easier to share them with the others

Reply:
Adding now @kasperpawlowski

Reply:
Implemented the pair of result arrays, sorting of the objects. Working on the output file option now.

Reply:
I think we should always dump to text file or json whether requested or not and we can gitignore the file as the results might not all show in the console